### ISSUE TITLE
Password reset is not sending emails

### ISSUE THREAD

**Issue by gig13** [owncloud/core#21561]
### Steps to reproduce
1. When a wrong password is submitted, click on the "Wrong password, reset it?"
2. 
3.

### Expected behaviour
When the user clicks the rest link, an email should be sent.

### Actual behavior
When clicking "Wrong password, reset it?" the following error (Couldn't
send reset email. Please make sure your username is correct.) is shown
and no email is sent.

### Server configuration
**Operating system**: RHEL 

**Web server:** Apache

**Database:** Oracle

**PHP version:** 5.4.40

**ownCloud version:** 8.2.1

**Updated from an older ownCloud or fresh install:** Updated

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' => '----',
  'passwordsalt' => '----',
  'trusted_domains' => 
  array (
    0 => 'owntest1.------',
    1 => 'owncloudtest.-----',
    2 => 'owntest2.----',
  ),
  'datadirectory' => '/var/www/html/owncloud/data',
  'dbtype' => 'oci',
  'version' => '8.2.1.4',
  'dbname' => 'owntest',
  'dbhost' => 'owndbtest.---',
  'dbtableprefix' => 'oc_',
  'dbtablespace' => 'users',
  'loglevel' => 3,
  'dbuser' => 'owncloud',
  'dbpassword' => 'REDACTED',
  'installed' => true,
  'license-key' => 'REDACTED',
  'log_type' => 'owncloud',
  'logfile' => '/var/log/owncloud/owncloud.log',
  'logtimezone' => 'America/Regina',
  'ldapIgnoreNamingRules' => false,
  'theme' => '',
  'maintenance' => false,
  'overwriteprotocol' => 'https',
  'mail_smtpmode' => 'php',
  'mail_smtphost' => 'REDACTED',
  'mail_smtpport' => '25',
  'session_lifetime' => 5400,
  'trashbin_retention_obligation' => '45, auto',
  'singleuser' => false,
  'appstoreenabled' => false,
  'has_internet_connection' => false,
  'ldapUserCleanupInterval' => 51,
  'supportedDatabases' => 
  array (
    0 => 'mysql',
    1 => 'pgsql',
    2 => 'oci',
  ),
  'secret' => 'REDACTED',
  'share_folder' => '/Shared',
  'asset-pipeline.enabled' => true,
  'mail_from_address' => 'owncloudtest',
  'mail_domain' => 'REDACTED',
  'memcache.local' => '\\OC\\Memcache\\APC',
  'mail_smtpsecure' => 'ssl',
  'mail_smtpauthtype' => 'NTLM',
);

```

**Are you using external storage, if yes which one:** LOCAL

**Are you using encryption:** NO

**Are you using an external user-backend, if yes which one:** AD

#### LDAP configuration 
on s3

#### ownCloud log
On s3

Email is working fine when I share a link, and I've tried both php and sendmail, please advise 

@MorrisJobke

**Comment by gig13** [comment]
On s3 at https://s3.owncloud.com/owncloud/index.php/apps/files?dir=%2FShared%2Fowncloud%2Fsupport%2Fgithub-issues%2Fcore%2F21561

report config
config.php
OC log snippet showing LDAPS errors

**Comment by MorrisJobke** [comment]
Is an email address set up for this user?

**Comment by gig13** [comment]
Yes, since share links work

**Comment by MorrisJobke** [comment]
> Yes, since share links work.

Ah okay. I just want to be super save. Not that we will try to debug quite hard and then have overseen such an easy problem.

**Comment by gig13** [comment]
@MorrisJobke Any updates?
I suggested to update to php55, but this appears to be a non starter due to RHEL6 support and our own documentation showing php54 installation for 8.2.

**Comment by MorrisJobke** [comment]
> But I will double check the LDAP mappings/configuration

If this is an LDAP instance, then the reset will never work. LDAP is read only. The link is removed for LDAP in https://github.com/owncloud/core/issues/12215 (8.2.3 and 9.0.0) or you specify an URL the user is forwarded to, to reset it's password in an LDAP facing interface.

@gig13 I hope this solves it for you.

**Comment by gig13** [comment]
@MorrisJobke 
I know the reset action will not work, but can an email at least be sent to the user (using the custom theme template - Lost password email) so that the Admins can redirect or provide appropriate guidance to end users.
"Your password needs to be reset by your Administrator, please contact admin@yourcompany.com"

**Comment by MorrisJobke** [comment]
> I know the reset action will not work, but can an email at least be sent to the user (using the custom theme template - Lost password email) so that the Admins can redirect or provide appropriate guidance to end users.
> "Your password needs to be reset by your Administrator, please contact admin@yourcompany.com"


This is what the lost_password_link was planned to be used for - see https://github.com/owncloud/core/issues/12215#issuecomment-110258578

**Comment by MorrisJobke** [comment]
@gig13 Does this work for you?

**Comment by gig13** [comment]
I will check with the originator

**Comment by MorrisJobke** [comment]
> Which OC version is this parameter in?

8.2.3 (unreleased but next maintenance release) & 9.0.0

**Comment by MorrisJobke** [comment]
As this is solved with 8.2.3 I will close this one here. If the solution doesn't work as intended then please open a separate ticket that describes, what is working in a wrong way. Thanks
